sdk/java-sdk/src/main/java/com/akkaserverless/javasdk/JwtClaims.java

```diff
@@ -0,0 +1,418 @@
+              }
+            });
+  }
+
+  /**
+   * Get the long claim with the given name.
+   *
+   * @param name The name of the claim.
+   * @return The long claim, if present. Returns empty if the claim is not a long or can't be parsed
+   *     as an long.
+   */
+  default Optional<Long> getLong(String name) {
+    return getString(name)
+        .flatMap(
+            value -> {
+              try {
+                return Optional.of(Long.parseLong(value));
+              } catch (NumberFormatException e) {
+                return Optional.empty();
+              }
+            });
+  }
+
+  /**
+   * Get the double claim with the given name.
+   *
+   * @param name The name of the claim.
+   * @return The double claim, if present. Returns empty if the claim is not a double or can't be
+   *     parsed as an double.
+   */
+  default Optional<Double> getDouble(String name) {
+    return getString(name)
+        .flatMap(
+            value -> {
+              try {
+                return Optional.of(Double.parseDouble(value));
+              } catch (NumberFormatException e) {
+                return Optional.empty();
+              }
+            });
+  }
+
+  /**
+   * Get the boolean claim with the given name.
+   *
+   * @param name The name of the claim.
+   * @return The boolean claim, if present. Returns empty if the claim is not a boolean or can't be
+   *     parsed as a boolean.
+   */
+  default Optional<Boolean> getBoolean(String name) {
+    return getString(name)
+        .flatMap(
+            value -> {
+              if (value.equalsIgnoreCase("true")) {
+                return Optional.of(Boolean.TRUE);
+              } else if (value.equalsIgnoreCase("false")) {
+                return Optional.of(Boolean.FALSE);
+              }
+              return Optional.empty();
+            });
+  }
+
+  /**
+   * Get the numeric data claim with the given name.
+   *
+   * <p>Numeric dates are expressed as a number of seconds since epoch, as described in RFC 7519
+   * section 2.
+   *
+   * @param name The name of the claim.
+   * @return The numeric date claim, if present. Returns empty if the claim is not a numeric date or
+   *     can't be parsed as a numeric date.
+   * @see <a href="https://datatracker.ietf.org/doc/html/rfc7519#section-2">RFC 7519 section 2</a>
+   */
+  default Optional<Instant> getNumericDate(String name) {
+    return getLong(name).map(Instant::ofEpochSecond);
+  }
+
+  /**
+   * Get the object claim with the given name.
+   *
+   * <p>This returns the claim as a Jackson JsonNode AST.
+   *
+   * @param name The name of the claim.
+   * @return The object claim, if present. Returns empty if the claim is not an object or can't be
+   *     parsed as an object.
+   */
+  default Optional<JsonNode> getObject(String name) {
+    return getString(name)
+        .flatMap(
+            value -> {
+              try {
+                return Optional.of(new ObjectMapper().readTree(value));
```

Comment:
Not sure it matters when it is only reading the tree, but, perhaps there some value to re-using the already existing, configured object mapper in `com.akkaserverless.javasdk.JsonSupport` for this rather than a new one for every call?

Reply:
Agreed that there is value to doing that, will change.